articles/container-registry/tasks-agent-pools.md

```diff
@@ -93,19 +93,38 @@ az acr agentpool update \
 
 Task agent pools require access to the following Azure services. The following firewall rules must be added to any existing network security groups or user-defined routes.
 
-| Direction | Protocol | Source         | Source Port | Destination          | Dest Port | Used    |
-| --------- | -------- | -------------- | ----------- | -------------------- | --------- | ------- |
-| Outbound  | TCP      | VirtualNetwork | Any         | AzureKeyVault        | 443       | Default |
-| Outbound  | TCP      | VirtualNetwork | Any         | Storage              | 443       | Default |
-| Outbound  | TCP      | VirtualNetwork | Any         | EventHub             | 443       | Default |
-| Outbound  | TCP      | VirtualNetwork | Any         | AzureActiveDirectory | 443       | Default |
-| Outbound  | TCP      | VirtualNetwork | Any         | AzureMonitor         | 443,12000 | Default |
+| Direction | Protocol | Source         | Source Port | Destination          | Dest Port | Used    | Remarks                                           |
+| --------- | -------- | -------------- | ----------- | -------------------- | --------- | ------- | ------------------------------------------------- |
+| Outbound  | TCP      | VirtualNetwork | Any         | AzureKeyVault        | 443       | Default |                                                   |
+| Outbound  | TCP      | VirtualNetwork | Any         | Storage              | 443       | Default |                                                   |
+| Outbound  | TCP      | VirtualNetwork | Any         | EventHub             | 443       | Default |                                                   |
+| Outbound  | TCP      | VirtualNetwork | Any         | AzureActiveDirectory | 443       | Default |                                                   |
+| Outbound  | TCP      | VirtualNetwork | Any         | AzureMonitor         | 443,12000 | Default | Port 12000 is a unique port used for diagnostics  |
 
 > [!NOTE]
 > If your tasks require additional resources from the public internet, add the corresponding rules. For example, additional rules are needed to run a docker build task that pulls the base images from Docker Hub, or restores a NuGet package.
 
 Customers basing their deployments with MCR can refer to [MCR/MAR firewall rules.](https://github.com/microsoft/containerregistry/blob/main/docs/client-firewall-rules.md)
 
+#### Advanced Network Configuration
+
+If the standard Firewall/NSG rules are too permissive and fine-grained control is required for outbound connections or if routing tables and network appliances are configured with the subnet, than the alternative is to enable service endpoints on the agent pool subnet in order to allow the agent pool access to its service dependencies. Outbound Firewall/NSG rules are still required in order for the Virtual Network to switch the source IP from a public to private IP in addition to the service endpoints being enabled.
```

Comment:
If the standard Firewall/NSG (Network Security Group) rules are deemed too permissive, and more fine-grained control is required for outbound connections, consider the following approach:

- Enable service endpoints on the agent pool subnet. This grants the agent pool access to its service dependencies while maintaining a secure network posture.
- It's important to note that outbound Firewall/NSG rules are still necessary. These rules facilitate the Virtual Network's ability to switch the source IP from public to private, which is an additional step beyond enabling service endpoints.